### PR TITLE
Add nod08a

### PR DESCRIPTION
**Pull request by ghost**
Add nod08a - "Battle for Zaire (a)"
http://cnc.wikia.com/wiki/Battle_for_Zaire

### CODE REVIEW

File: mods/cnc/maps/nod08a/nod08a.lua [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+					SendHelicopter()
+			end)
+		end
+	end)
+end
+
+SendAttackWave = function(team)
+	for type, amount in pairs(team.units) do
+		count = 0
+		actors =  enemy.GetActorsByType(type)
```
Comment (abcdefg30) [inline mods/cnc/maps/nod08a/nod08a.lua]: Duplicated space after the `=`.

File: mods/cnc/maps/nod08a/nod08a.lua [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+	end
+end
+
+SendGDIAirstrike = function(hq, delay)
+	if not hq.IsDead and hq.Owner == enemy then
+		local target = getAirstrikeTarget()
+
+		if target then
+			hq.SendAirstrike(target, false, Facing.NorthEast + 4)
+			Trigger.AfterDelay(delay, function() SendGDIAirstrike(hq, delay)  end)
```
Comment (abcdefg30) [inline mods/cnc/maps/nod08a/nod08a.lua]: Two spaces after `SendGDIAirstrike(hq, delay)` here and two lines below.

File: mods/cnc/maps/nod08a/nod08a.lua [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+Trigger.OnAllKilledOrCaptured(Outpost, function()
+	if not outpostCaptured then
+		outpostCaptured = true
+
+		Trigger.AfterDelay(DateTime.Minutes(1), function()
+			airstrikeproxy = Actor.Create("airstrike.proxy", false, { Owner = enemy })
+
+			if not GDIHQ.IsDead and (not NodHand.IsDead or not NodNuke.IsDead) then
+				airstrikeproxy.SendAirstrike(AirstrikeTarget.CenterPosition, false, Facing.NorthEast + 4)
```
Comment (abcdefg30) [inline mods/cnc/maps/nod08a/nod08a.lua]: Can we move the actor creation etc into the check as well?
```
if not GDIHQ.IsDead and (not NodHand.IsDead or not NodNuke.IsDead) then 
	local airstrikeproxy = Actor.Create("airstrike.proxy", false, { Owner = enemy }) 
	airstrikeproxy.SendAirstrike(AirstrikeTarget.CenterPosition, false, Facing.NorthEast + 4) 
	airstrikeproxy.Destroy() 
end
```
Also notice the `local` keyword. As we don't want to reference the airstrike actor from elsewhere again
I think we don't need to have this global.

File: mods/cnc/maps/nod08a/nod08a.lua [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+			Trigger.AfterDelay(delay, function() SendGDIAirstrike(hq, delay)  end)
+		else
+			Trigger.AfterDelay(delay/4, function() SendGDIAirstrike(hq, delay)  end)
+		end
+	end
+end
+
+SendWaves = function(counter, Waves)
+	if counter <= #Waves then
+		team = Waves[counter]
```
Comment (abcdefg30) [inline mods/cnc/maps/nod08a/nod08a.lua]: `local team = Waves[counter]`, please.

**Comment by abcdefg30** [comment]
Looks good to me. :+1: / :white_check_mark: Nicely done!

One thing though: In the beginning you can easily outrange most of the GDI rifle man with your rocket soldiers. Could we get an `OnDamaged` trigger for those few infantry units to have them fight back?

**Comment by Biofreak1987** [comment]
Added onDamaged Trigger for NodBase / Outpost Guards and reworked GDI Orcas.

File: mods/cnc/maps/nod08a/nod08a.lua [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+
+GuardNodBase = function()
+	Utils.Do(IntroGuards, function(guard)
+		Trigger.OnDamaged(guard, function(guard)
+			IdleHunt(guard)
+		end)
+	end)
+end
+
+GuardOutpost = function()
```
Comment (Mailaender) [inline mods/cnc/maps/nod08a/nod08a.lua]: This is a bit of code duplication. You could make this

```lua
AutoGuard = function(guards)
	Utils.Do(guards, function(guard)
		Trigger.OnDamaged(guard, function(guard)
			IdleHunt(guard)
		end)
	end)
end

AutoGuard(IntroGuards)
AutoGuard(OutpostGuards)
```

**Comment by Mailaender** [comment]
Looks good otherwise. Nice attention to detail with the Orcas defending the base.

**Comment by Biofreak1987** [comment]
Fixed

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/7c659c07478ec3d39bdc631c29f3390e45a488b0...2415d61f9265411d363f42ffa01114b67e87bdac)

**Pull request merged**